From 2105bbdfcf11082765daba013b64f1c37aaf6245 Mon Sep 17 00:00:00 2001
From: Pascal Quantin <pascal.quantin@gmail.com>
Date: Mon, 3 Sep 2012 12:48:57 +0000
Subject: [PATCH 3/4] Fix
 https://bugs.wireshark.org/bugzilla/show_bug.cgi?id=7666
 : Check that DRDA command has a minimum length of 10
 bytes to prevent a potential infinite loop

svn path=/trunk/; revision=44749
---
 epan/dissectors/packet-drda.c |    5 +++++
 1 file changed, 5 insertions(+)

diff --git a/epan/dissectors/packet-drda.c b/epan/dissectors/packet-drda.c
index f5d701b..730f687 100644
--- a/epan/dissectors/packet-drda.c
+++ b/epan/dissectors/packet-drda.c
@@ -55,6 +55,7 @@
 #include <epan/packet.h>
 #include <epan/conversation.h>
 #include <epan/prefs.h>
+#include <epan/expert.h>
 #include "packet-tcp.h"
 
 static int proto_drda = -1;
@@ -696,6 +697,10 @@ dissect_drda(tvbuff_t *tvb, packet_info *pinfo, proto_tree *tree)
     {
         iCommand = tvb_get_ntohs(tvb, offset + 8);
         iLength = tvb_get_ntohs(tvb, offset + 0);
+        if (iLength < 10) {
+            expert_add_info_format(pinfo, NULL, PI_MALFORMED, PI_ERROR, "Invalid length detected (%u): should be at least 10 bytes long", iLength);
+            break;
+        }
         /* iCommandEnd is the length of the packet up to the end of the current command */
         iCommandEnd += iLength;
 
-- 
1.7.10.4

